### ISSUE TITLE
Add variables to env_info during installation

### ISSUE THREAD

**Issue by rweickelt** [conan-io/conan#4493]
**TLDR**: Package A depends on B. I call ``conan install . -g virtualenv`` on A and I want activate.sh to contain a variable BLABLUB=<PACKAGE_INSTALL_PATH_OF_PACKAGE_B>. This variable is not exported by B and I cannot modify B, so I have to create it somehow in the conanfile of A. How can I do that?

----

In an Conan recipe, I want to set some environment variables so that they become available in the virtual environment created for the same package.

The variable depends on the installation path of a dependency. I tried to modify it during configure(), but that doesn't work:

```
def configure(self):
        self.env_info.BLABLA_PATH = self.deps_env_info["somepackage"].rootpath
```

Is there a way to achieve that?

**Comment by Johnnyxy** [comment]
If I am not mistaken at the configure stage the dependency graph has not been computed yet. This implies that one cannot access the dependency informations of any dependency.

If you need to set environment variables at build time you can do this via normal [python commands](https://docs.python.org/3.6/library/os.html#os.putenv).

If you need the information at build time (e.g. execute a tool that you have as dependency) you can do this automatically ([RunEnvironment](https://docs.conan.io/en/latest/reference/build_helpers/run_environment.html)):
```python
def build(self):
    env_build = RunEnvironment(self)
    with tools.environment_append(env_build.vars):
        cmake.configure(...)
        cmake.build()
```

**Comment by rweickelt** [comment]
Thanks for your reply. But I want those variables to be available after conan install -g virtualenv has generated the activate.sh.

**Comment by Johnnyxy** [comment]
You want to change the generated environment after virtualenv ran.

So recipe A depends on recipe B. 
Virtualenv generated all env vars for B.

What do you try to achieve now? What is the use case?

**Comment by rweickelt** [comment]
A depends on B, but I don't have control over B. B is missing some variables that I need when working with A.  In this case I am using Conan for package management, not as meta build system. I only call ``conan install -g virtualenv`` to install dependencies and to create a virtual environment. Does that make sense?

**Comment by Johnnyxy** [comment]
Ah ok. You could define your line 
```python
self.env_info.BLABLA_PATH = self.deps_env_info["somepackage"].rootpath
```
in the ```package_info(self)``` function in A. Then you could "install" recipe A with virtualenv what in turn would merge env vars from B and A (in theory :)).

**Comment by rweickelt** [comment]
Thanks for helping me, but I am afraid this wouldn't work. When "installing" A, then the ``package_info()`` function of A is not called. I think this by design. ``package_info()`` is supposed to `export` information to package consumers only.

I am currently bound to conan 1.6.1, I don't know if that has changed since then.

**Comment by Johnnyxy** [comment]
Hm, the docs say package_info is executed when you call [conan install](https://docs.conan.io/en/1.6/reference/commands/consumer/install.html#conan-install).

**Comment by Johnnyxy** [comment]
Maybe consider creating another package.
A (lacks of env vars)
B depends on A (add serval env vars)
C depends on B (conan install this one)

**Comment by rweickelt** [comment]
Yes, ``package_info()`` is called, but only in the dependency packages, not for the package I want to work with.

Thanks for suggesting a work-around. Can I solve that in a single recipe file? 

I can't believe that is not possible out-of-the-box. My use-case doesn't sound that odd.

**Comment by Johnnyxy** [comment]
For that you have to wait for someone who is more involved in conan :). 
@uilianries

**Comment by lasote** [comment]
`conan export. <REF_OF_A> && conan install <REF_OF_A>` should work

**Comment by rweickelt** [comment]
@lasote , could you explain how that addresses my problem?

**Comment by rweickelt** [comment]
I looked in the source code and it appears, that nothing is called in the root Conan file that would allow me to inject additional information in the generator during the install step AFTER dependencies have been resolved. So there is no good solution. Might be a good feature to add.

**Comment by memsharded** [comment]
The environment has been fully revamped:
- ``[buildenv]`` in profiles
- ``self.buildenv_info`` in recipes
- New generators ``VirtualBuildEnv``
- No injection of environment in recipe methods, only in ``self.run()`` calls

Closing this as outdated, please open new tickets as necessary for any further questions or issues.
